odoo/cli/i18n.py

```diff
@@ -0,0 +1,228 @@
+                # bit overkill, but better safe than sorry
+                csv.field_size_limit(500 << 20)
+
+        export_pot = 'pot' in parsed_args.languages
+        if export_pot:
+            if parsed_args.output and parsed_args.output.endswith('.csv'):
+                self.export_parser.error(
+                    "Cannot export template in .csv format, please specify a language.")
+            parsed_args.languages.remove('pot')
+
+        with Registry.new(parsed_args.db_name).cursor() as cr:
+            env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
+
+            # We want to log invalid parameters
+            modules = env['ir.module.module'].with_context(active_test=False).search_fetch(
+                [('name', 'in', parsed_args.modules)], ['name', 'state'])
+            if not_found_module_names := set(parsed_args.modules) - set(modules.mapped("name")):
+                _logger.warning("Ignoring not found modules: %s", ", ".join(not_found_module_names))
+            if not_installed_modules := modules.filtered(lambda x: x.state != 'installed'):
+                modules_str = ", ".join(not_installed_modules.mapped("name"))
+                _logger.warning("Ignoring not installed modules: %s", modules_str)
+                modules -= not_installed_modules
+            if len(modules) < 1:
+                self.export_parser.error("No valid module has been provided")
+            module_names = modules.mapped("name")
+
+            languages = self._get_languages(env, parsed_args.languages)
+            languages_count = len(languages) + (1 if export_pot else 0)
+            if languages_count == 0:
+                self.export_parser.error("No valid language has been provided")
+
+            if parsed_args.output == '-':
+                # To standard output
+                trans_export(languages.code, module_names, sys.stdout.buffer, 'po', env.cr)
+            elif parsed_args.output:
+                # To specified single output file
+                self._export_file(env, module_names, languages.code, Path(parsed_args.output))
+            else:
+                # Po(t) files in the modules' i18n folders
+                for module_name in module_names:
+                    i18n_path = Path(get_module_path(module_name)) / 'i18n'
+                    if export_pot:
+                        path = i18n_path / f'{module_name}.pot'
+                        self._export_file(env, [module_name], None, path)
+                    for language in languages:
+                        path = i18n_path / f'{language.iso_code}.po'
+                        self._export_file(env, [module_name], language.code, path)
+
+    def _export_file(self, env, module_names, lang_code, path):
+        source = module_names[0] if len(module_names) == 1 else 'modules'
+        _logger.info("Exporting %s (%s) to %s", source, lang_code or 'template', path)
+        if not path.parent.exists():
+            os.makedirs(path.parent)
```

Comment:
single system call and thread-safe (even if we don't need thread safety here)
```suggestion
        path.parent.mkdir(exist_ok=True)
```